tensorflow/core/tpu/tpu_initializer_helper.cc

```diff
@@ -65,16 +65,15 @@ bool IsTpuUsed(long pid) {
   }
   std::unique_ptr<DIR, int (*)(DIR*)> fd_dir(raw_fd_dir, closedir);
   struct dirent* ent;
-  char line[100];
+  std::string line;
   std::string tpu_dev_path = "/dev/accel0";
   while (ent = readdir(raw_fd_dir)) {
     if (!isdigit(*ent->d_name)) continue;
     long fd = strtol(ent->d_name, NULL, 10);
     path = absl::StrCat("/proc/", pid, "/fd/", fd);
-    if (!readlink(path.c_str(), line, sizeof(line))) continue;
-    std::string str_line(line);
-    str_line.resize(tpu_dev_path.size());
-    if (str_line != tpu_dev_path) continue;
+    if (!readlink(path.c_str(), &line[0], line.size())) continue;
+    line.resize(tpu_dev_path.size());
```

Comment:
Move this below L68, so line is always size tpu_dev_path. Otherwise it starts as size 0, and you won't actually read anything!

Reply:
hmmm, actually both ways worked for me surprisingly! but I moved it up to make it cleaner.

Reply:
It may have worked since it got resized after the first try, and it's unlikely for the fd you're looking to be the first one.